### PR TITLE
EMTF April 2018 emulator update part 2 - GEM and Track Trigger data formats

### PR DESCRIPTION
**Pull request by abrinke1**
Part 2 of a series of PRs from @jiafulow (following #22820 ) which will bring the EMTF emulator up-to-date for 2018 running.  This PR adds data formats for Run 3 and Phase II trigger primitives in the endcaps, and creates a framework for these TPs to be processed by the EMTF emulator.  These changes will benefit various groups developing L1T muon for post-2020 running, including @dildick , @mmaggi , @camilocarrillo , @gabrielhpugliese , and others.

This PR adds no new functionalities on its own, and changes nothing about how the EMTF emulator runs.  It only provides foundational elements (data formats and TP processing framework) for following PRs, in which TP processing in EMTF will be actually enabled when running a "future" scenario.  @rekovic , @thomreis please take a look.

Best regards,
Andrew

### CODE REVIEW

**Comment by thomreis** [comment]
please test

**Comment by gabrielhpugliese** [comment]
You have probably mentioned me by mistake

**Comment by abrinke1** [comment]
Hi @gabrielhpugliese ,
Sorry about that!  I meant "Gabriella Pugliese" ...
Cheers, Andrew

File: L1Trigger/L1TMuonEndCap/src/TTPrimitiveConversion.cc [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+void TTPrimitiveConversion::process(
+    const std::map<int, TTTriggerPrimitiveCollection>& selected_ttprim_map,
+    EMTFHitCollection& conv_hits
+) const {
+  std::map<int, TTTriggerPrimitiveCollection>::const_iterator map_tp_it  = selected_ttprim_map.begin();
+  std::map<int, TTTriggerPrimitiveCollection>::const_iterator map_tp_end = selected_ttprim_map.end();
+
+  for (; map_tp_it != map_tp_end; ++map_tp_it) {
```
Comment (rekovic) [inline L1Trigger/L1TMuonEndCap/src/TTPrimitiveConversion.cc]: Should be able to simplify and use auto-range for loops 
```
for (auto& m: selected_ttprim_map) {

  for( auto& tp : m.second() {
```

File: L1Trigger/L1TMuonEndCap/src/TTPrimitiveConversion.cc [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+void TTPrimitiveConversion::process_no_prim_sel(
+    const TTTriggerPrimitiveCollection& ttmuon_primitives,
+    EMTFHitCollection& conv_hits
+) const {
+  TTTriggerPrimitiveCollection::const_iterator tp_it  = ttmuon_primitives.begin();
+  TTTriggerPrimitiveCollection::const_iterator tp_end = ttmuon_primitives.end();
+
+  for (; tp_it != tp_end; ++tp_it) {
```
Comment (rekovic) [inline L1Trigger/L1TMuonEndCap/src/TTPrimitiveConversion.cc]: Please make this auto-range for loop with
`for (auto& tp : TTTriggerPrimitiveCollection)`

No need for `tp_it` and `tp_end` above.

**Comment by thomreis** [comment]
please test

**Comment by rekovic** [comment]
+1

**Comment by fabiocos** [comment]
@abrinke1 do I understand correctly that for the time being this code is not really run in a test workflows? In this case, how has this been tested?

**Comment by fabiocos** [comment]
@abrinke1 ok, this is what I meant, no effect is observed because so far most of this code is not really exercised in tests

**Comment by fabiocos** [comment]
+1

**Pull request merged**